src/VisualStudio/Core/Def/Implementation/Progression/GraphQueries/InheritedByGraphQuery.cs

```diff
@@ -32,7 +27,7 @@ internal sealed class InheritedByGraphQuery : IGraphQuery
                 if (namedType.TypeKind == TypeKind.Class)
                 {
                     var derivedTypes = await namedType.FindDerivedClassesAsync(solution, cancellationToken).ConfigureAwait(false);
-                    foreach (var derivedType in derivedTypes)
+                    foreach (var derivedType in derivedTypes.Where(derived => derived.BaseType == namedType))
```

Comment:
We don't have another helper for this somewhere? I feel bad we're doing lots of work chasing types only to throw them out.